modules/core/shared/src/main/scala/gem/math/JulianDate.scala

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.math
+
+import cats.{Eq, Show}
+
+import java.time.{Instant, LocalDateTime}
+import java.time.ZoneOffset.UTC
+
+/** Astronomical time representation of continuous days since noon, November 24,
+  * 4714 BC.
+  *
+  * @param dayNumber      whole Julian Day number
+  * @param nanoAdjustment fraction of a day adjustment in nanoseconds
+  *
+  * @see The Wikipedia [[https://en.wikipedia.org/wiki/Julian_day article]]
+  */
+sealed abstract case class JulianDate(
```

Comment:
Should you override equals/hashCode in this case?

Reply:
No, you still get a generated `equals/hashCode` but no `copy` and no companion `apply`.

Reply:
so `abstract case class` just forbids construction but the rest of the case class sematics apply?